adapters/handlers/rest/configure_api.go

```diff
@@ -374,6 +375,35 @@ func MakeAppState(ctx context.Context, options *swag.CommandLineOptionsGroup) *s
 		remoteIndexClient, appState.Logger, appState.ServerConfig.Config.Persistence.DataPath)
 	appState.Scaler = scaler
 
+	// let classTenantDataEvents be nil if the metadata server is not enabled since the metadata
+	// server/querierManager are the users of the channel
+	var classTenantDataEvents chan metadataserver.ClassTenant
+	querierManager := metadataserver.NewQuerierManager(appState.Logger)
+	metadataServer := metadataserver.NewServer(
+		appState.ServerConfig.Config.MetadataServer.GrpcListenAddress,
+		1024*1024*1024,
+		true,
+		querierManager,
+		appState.ServerConfig.Config.MetadataServer.DataEventsChannelCapacity,
+		appState.Logger)
+	appState.MetadataServer = metadataServer
+
+	if appState.ServerConfig.Config.MetadataServer.Enabled {
+		classTenantDataEvents = make(chan metadataserver.ClassTenant, appState.ServerConfig.Config.MetadataServer.DataEventsChannelCapacity)
+		enterrors.GoWrapper(func() {
+			err := metadataServer.Open()
+			if err != nil {
+				appState.Logger.Errorf("metadata server did not start: %v", err)
+			}
+		}, appState.Logger)
+
+		enterrors.GoWrapper(func() {
+			for classTenantDataEvent := range classTenantDataEvents {
+				querierManager.NotifyClassTenantDataEvent(classTenantDataEvent)
```

Comment:
curious what happens if `querierManager.NotifyClassTenantDataEvent(classTenantDataEvent)` returns error?.  Because `GoWrapper` [doesn't look like handles](https://github.com/weaviate/weaviate/blob/main/entities/errors/go_wrapper.go#L36) the error returned from `NotifyClassTenantDataEvent`. 

May be I miss something?

Reply:
Thanks, just forgot to use the error after I added it